File: pkgs/mrlyrs/mrlymath/src/six/painter.rs
use super::models::Cell6d;
use mrlycore::cell::mapping;
use mrlycore::colors::Color;
use mrlycore::enums::Mode;
use std::collections::HashMap;

pub fn paint(
    mut cell: Cell6d,
    custom: Option<&HashMap<u8, Vec<Color>>>,
    mode: Option<Mode>,
) -> Cell6d {
    let defaults = mapping();
    let mapping = custom.unwrap_or(&defaults);
    cell.cell = cell.cell.paint(mapping, mode.unwrap_or(Mode::Type));
    cell
}

File: apps/jsx/src/mrly/six/painter.ts
import type { Mode } from "../enums";
import type { PaletteMap } from "../two/models";
import { paint2d } from "../two/painter";
import type { Cell6d } from "./models";

export function paint6d(cell: Cell6d, palette?: PaletteMap, mode?: Mode): Cell6d {
  return paint2d(cell, palette, mode) as Cell6d;
}
